crates/solver/src/settlement_ranker.rs

```diff
@@ -194,33 +194,59 @@ impl SettlementRanker {
                         .await;
                     (solver, simulation)
                 },
-            ))
-            .await
-            .into_iter()
-            .filter_map(|(solver, result)| match result {
-                Ok(res) => Some((solver, res)),
-                Err(err) => {
-                    tracing::warn!(?err, "error in settlement rating logic");
+            ),
+        )
+        .await
+        .into_iter()
+        .filter_map(|(solver, result)| match result {
+            Ok(res) => Some((solver, Rating::Ok(res))),
+            Err(err) => match err {
+                RatingError::FailedSimulation(error) => Some((solver, Rating::Err(error))),
+                RatingError::FailedScoring(error) => {
+                    tracing::debug!(
+                        solver_name = %solver.name(), ?error,
+                        "settlement filtered",
+                    );
+                    let reason = match error {
+                        ScoringError::ObjectiveValueNonPositive(_) => {
+                            Some(SolverRejectionReason::ObjectiveValueNonPositive)
+                        }
+                        ScoringError::SuccessProbabilityOutOfRange(_) => {
+                            Some(SolverRejectionReason::SuccessProbabilityOutOfRange)
+                        }
+                        ScoringError::ScoreHigherThanObjective(_) => {
+                            Some(SolverRejectionReason::ScoreHigherThanObjective)
+                        }
+                        ScoringError::InternalError(_) => None,
+                    };
+                    if let Some(reason) = reason {
+                        solver.notify_auction_result(auction_id, AuctionResult::Rejected(reason));
```

Comment:
Why are we not notifying solvers in the case of the simulation failing or other errors? I assume we do notify for simulation failures already somewhere else in the code, but where and can we consolidate the notification logic to make it harder to miss reasons to notify?

Reply:
I moved the notify for simulation https://github.com/cowprotocol/services/pull/1877/commits/b3629a1787df03864413c51cc61224f920708c33

I think there is no point leaking out our internal errors to solvers, such as division by zero or failed web3 communication... There is nothing they can do about it and it would probably just raise questions (what kind of errror is it etc)